app/web/features/translate/LanguagePickerSelect.tsx

```diff
@@ -80,6 +80,9 @@ export default function LanguagePickerSelect({
       await changeLanguageMutation(newLocale);
     }
 
+    // Set the NEXT_LOCALE cookie
+    document.cookie = `NEXT_LOCALE=${newLocale}; path=/; max-age=31536000; SameSite=Lax`;
```

Comment:
If you are sending this to the backend, then it should be setting the cookie itself